### PR TITLE
Middleware endpoints

### PR DESCRIPTION
**Pull request by dbartalino**
Removed /:vin per comments
Temporarily removed endpoints to edit vehicles and pull vehicle by VIN

### CODE REVIEW

File: middleware/app.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 const express = require('express');
 const cors = require('cors');
+const bodyParser = require('body-parser');
+//const http = require('http');
```
Comment (dsifford) [inline middleware/app.js]: Delete me.

File: middleware/app.js [hunk 2]
```diff
@@ -26,8 +29,45 @@ app.delete('/', function(_, res) {
   res.send(`DELETE request homepage - localhost:${PORT}`);
 });
 
+// List all vehicles from blockchain
+app.get('/vehicles', function(req, res) {
+  (async () => {
```
Comment (dsifford) [inline middleware/app.js]: Don't do async this way. Either make the entire function async or none of it at all.

File: middleware/app.js [hunk 3]
```diff
@@ -26,8 +29,45 @@ app.delete('/', function(_, res) {
...
+  
+    const resVehicleList = await client.listVehicles();
+    console.log(resVehicleList);
+    res.send(resVehicleList);
+  })();
+});
+
+// Post a new vehicle to the blockchain
+app.post('/vehicles', function(req, res) {
+  (async () => {
```
Comment (dsifford) [inline middleware/app.js]: Don't do async this way. Either make the entire function async or none of it at all.

File: processor/.dockerignore [hunk 4]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (dsifford) [inline processor/.dockerignore:1]: Why is this needed?

**Comment by dsifford** [comment]
I disagree with @ej3076. I think vin should be passed in on the url if looking up a single vehicle. That's how REST should work.

File: middleware/app.js [hunk 5]
```diff
@@ -26,8 +29,45 @@ app.delete('/', function(_, res) {
...
+    res.send(resPostVehicle);
+  })();
+});
+
 app.use('/keys', require('./routes/keys'));
 
 app.listen(PORT, () =>
   console.log(`Server listening on port ${IS_PRODUCTION ? PORT : '8080'}.`),
-);
+);
```
Comment (sloan-seamus) [inline middleware/app.js]: @ej3076 yelled at me for not having white space at the end of my file, so now I have to do it to you too lol
